internal/magic/magic.go

```diff
@@ -104,7 +104,13 @@ func xmlCheck(sig xmlSig, raw []byte) bool {
 // matches the raw input.
 func markup(sigs ...[]byte) Detector {
 	return func(raw []byte, limit uint32) bool {
-		raw = trimLWS(raw)
+		if bytes.HasPrefix(raw, []byte{0xEF, 0xBB, 0xBF}) {
+			// we restore BOM after stripping WS so its presence can be used
+			// in subsequent functions
+			raw = append([]byte{0xEF, 0xBB, 0xBF}, trimLWS(raw[3:])...)
```

Comment:
This line allocates another copy of the `raw` bytes slice and that could be slow/consume too much memory, depending on `raw` size.